src/tabs/tab-header-bar.tsx

```diff
@@ -107,6 +138,18 @@ export function TabHeaderBar({
     }
   };
 
+  useEffect(() => {
```

Comment:
Mutating the DOM in `useEffect` can cause many problems with inconsistency, so we should avoid that if at all possible. Focusing probably does need to be in here, but setting the `tabIndex` should be done in the standard render logic.